client/src/document/organisms/toc/placement.tsx

```diff
@@ -0,0 +1,149 @@
+import { useCallback, useEffect, useRef } from "react";
+import { useIsServer } from "../../../hooks";
+import {
+  PlacementStatus,
+  usePlacementStatus,
+} from "../../../placement-context";
+import { useUserData } from "../../../user-context";
+import { usePageVisibility } from "../../../writers-homepage/hooks";
+
+interface Timer {
+  timeout: number | null;
+  start: number | null;
+  notVisible?: boolean;
+}
+
+function viewed(
+  pong: PlacementStatus,
+  observer: IntersectionObserver | null = null
+) {
+  navigator?.sendBeacon(
+    `/pong/viewed?code=${encodeURIComponent(pong?.view)}${
+      pong?.fallback
+        ? `&fallback=${encodeURIComponent(pong?.fallback?.view)}`
+        : ""
+    }`
+  );
+  observer?.disconnect();
+}
+
+export function Placement() {
+  const isServer = useIsServer();
+  const user = useUserData();
+  const pong = usePlacementStatus();
+  const isVisible = usePageVisibility();
+
+  const observer = useRef<IntersectionObserver | null>(null);
+  const timer = useRef<Timer>({ timeout: null, start: null });
+
+  const place = useCallback(
+    (node) => {
+      if (pong && node !== null && !observer.current) {
+        const observerOptions = {
+          root: null,
+          rootMargin: "0px",
+          threshold: [0.5],
+        };
+        const intersectionObserver = new IntersectionObserver((entries) => {
+          console.log(entries);
+          const [{ isIntersecting = false, intersectionRatio = 0 } = {}] =
+            entries;
+          if (
+            isIntersecting &&
+            intersectionRatio >= 0.5 &&
+            typeof navigator !== "undefined" &&
+            typeof window !== "undefined"
+          ) {
+            if (timer.current.timeout === null) {
+              timer.current = {
+                timeout: window.setTimeout(
```

Comment:
Most of the time, we just do `setTimeout()`:

```suggestion
                timeout: setTimeout(
```

Like this, we shouldn't need to check for `window !== "undefined"` above.